src/bases/PckDao.sol

```diff
@@ -272,15 +274,28 @@ abstract contract PckDao is DaoBase, SigVerifyBase {
      */
     function _getAllTcbs(bytes16 qeidBytes, bytes2 pceidBytes) internal view virtual returns (bytes18[] memory tcbms);
 
-    function _validatePck(CA ca, bytes memory der, bytes18 tcbm, bytes2 pceid) internal view returns (bytes32 hash) {
-        // Step 1: Check whether the pck has expired
-        bool notExpired = pckLib.certIsNotExpired(der);
+    function _validatePck(CA ca, bytes memory der, bytes16 qeid, bytes2 pceid, bytes18 tcbm) internal view returns (bytes32 hash, bytes32 key) {
+        X509CertObj memory pck = pckLib.parseX509DER(der);
+        
+        // Step 0: Check whether the pck has expired
+        bool notExpired = block.timestamp > pck.validityNotBefore && block.timestamp < pck.validityNotAfter;
         if (!notExpired) {
             revert Certificate_Expired();
         }
 
-        X509CertObj memory pck = pckLib.parseX509DER(der);
         hash = keccak256(pck.tbs);
+        key = PCK_KEY(qeid, pceid, tcbm);
+
+        // Step 1: Rollback prevention: new certificate should not have an issued date
+        // that is older than the existing certificate
+        bytes memory existingData = _fetchDataFromResolver(key, false);
+        if (existingData.length > 0) {
+            (uint256 existingCertNotValidBefore, ) = crlLib.getCrlValidity(existingData);
+            bool replaced = existingCertNotValidBefore > pck.validityNotBefore;
```

Comment:
Why is it called replaced here?

Reply:
[`6eb7099`](https://github.com/automata-network/automata-on-chain-pccs/pull/15/commits/6eb7099e3117aab71518fd1d9ea0cb64b0d2023a)

Reply:
[6eb7099](https://github.com/automata-network/automata-on-chain-pccs/pull/15/commits/6eb7099e3117aab71518fd1d9ea0cb64b0d2023a)